### ISSUE TITLE
Rancher upgrade fails with no healthy host found error

### ISSUE THREAD

**Issue by ippbuilder** [rancher/rancher#9833]
I have one AWS instance for development purposes and I am using Rancher server and agent both in this server. The MySQL is not inside rancher server (it is another container in same host). There are a few services (dockerized) running on this server and the first time I set up the rancher everything works fine and services are deployed all fine.

The problem happens when I need to upgrade service for any reason and from that moment all my upgrades (even on other services start to fail). I tried many solutions I found and I checked the error logs but nothing meaningful found. 

There is something interesting though. When "Scheduling" set to "Automatically pick hosts for each container matching scheduling rules" upgrade fails but if I set it manually to "Run all containers on a specific host" it doesn't fail and upgrade succeed. I think the error is totally misleading as it implies something is wrong with the host but it seems the scheduler is having some problem.

I would appreciate any help to set this right cause I was just started to like Rancher!

Thanks

In case it helps the OS is Ubuntu 16:04

What I see in Rancher UI: someservice (Failed to allocate instance [container:1i2917]: Bad instance [container:1i2917] in state [erroring]: Allocation failed: No healthy hosts meet the resource constraints: [809:80/tcp portReservation, instanceReservation: 1]) | Image:

What I see in Rancher container logs:

2017-09-06 03:43:54,827 ERROR [:] [] [] [] [cutorService-87] [.e.s.i.ProcessInstanceDispatcherImpl] Unknown exception running process [service.upgrade:46003] on [26], canceled by [State [removed] is not valid for process [instance.stop:46012] on resource [2736]] 

time="2017-09-06T03:54:52Z" level=error msg="Waiting for 1s21 failed: Failed to allocate instance [container:1i2754]: Bad instance [container:1i2754] in state [erroring]: Allocation failed: No healthy hosts meet the resource constraints: [805:80/tcp portReservation, instanceReservation: 1]" 

2017-09-06 07:18:56,639 ERROR [:] [] [] [] [utorService-105] [.e.s.i.ProcessInstanceDispatcherImpl] Unknown exception running process [service.upgrade:48154] on [21], canceled by [State [removed] is not valid for process [instance.stop:48168] on resource [2904]] 

time="2017-09-05T09:50:25Z" level=error msg="Error Action [upgrade] not available on [&{1s22 service map[account:http://localhost:8080/v2-beta/projects/1a5/services/1s22/account consumedbyservices:http://localhost:8080/v2-beta/projects/1a5/services/1s22/consumedbyservices consumedservices:http://localhost:8080/v2-beta/projects/1a5/services/1s22/consumedservices instances:http://localhost:8080/v2-beta/projects/1a5/services/1s22/instances networkDrivers:http://localhost:8080/v2-beta/projects/1a5/services/1s22/networkdrivers stack:http://localhost:8080/v2-beta/projects/1a5/services/1s22/stack storageDrivers:http://localhost:8080/v2-beta/projects/1a5/services/1s22/storagedrivers self:http://localhost:8080/v2-beta/projects/1a5/services/1s22 serviceExposeMaps:http://localhost:8080/v2-beta/projects/1a5/services/1s22/serviceexposemaps serviceLogs:http://localhost:8080/v2-beta/projects/1a5/services/1s22/servicelogs containerStats:http://localhost:8080/v2-beta/projects/1a5/services/1s22/containerstats] map[rollback:http://localhost:8080/v2-beta/projects/1a5/services/1s22/?action=rollback remove:http://localhost:8080/v2-beta/projects/1a5/services/1s22/?action=remove finishupgrade:http://localhost:8080/v2-beta/projects/1a5/services/1s22/?action=finishupgrade removeservicelink:http://localhost:8080/v2-beta/projects/1a5/services/1s22/?action=removeservicelink addservicelink:http://localhost:8080/v2-beta/projects/1a5/services/1s22/?action=addservicelink setservicelinks:http://localhost:8080/v2-beta/projects/1a5/services/1s22/?action=setservicelinks]}] in upgrading service 1s22" 

-- | --

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.6.7` Cattle: `v0.183.5` UI: `v1.6.17` |
|Access|`localauth` `admin`|
|Orchestration|`Cattle`|
|Route|`stacks.index`|

**Comment by aemneina** [comment]
@ippbuilder how big is that host, spec-wise? Is it doing well on resources? Any errors in the docker logs that you can share?

**Comment by cjellick** [comment]
When you select the "run all containers on a specific host" option, we force the containers to that host, even if there is a scheduling conflict.
When you don't select that option, we let the scheduler decide. The logs indicate that you want port 805, but no hosts have that port available.

**Comment by ippbuilder** [comment]
> how big is that host, spec-wise? Is it doing well on resources? Any errors in the docker logs that you can share?

The instance has 2 CPU and 8 GB memory which is more than enough to host a few small web app. I experienced the same setup with ECS and Docker Cloud which never ran our of resources. The are a huge load of errors and impossible to check but I did share some meaningful errors I saw.

> When you don't select that option, we let the scheduler decide. The logs indicate that you want port 805, but no hosts have that port available.

I did realize the problem but my question is if there is no host available with that port then it should not work (whether I ignore scheduler or not).

**Comment by sirius1024** [comment]
Same problem here. Dont know how to fix it.

**Comment by ruiztulio** [comment]
Hi guys, I jus started using rancher and I get into the same problem:

Rancher 1.6.14
Docker: 17.12.1-ce, build 7390fc6

I have Ranches in a VM with mysql in the host (the same host serve as node too) and another VM wich I use for the tests, I tried deploying in different ports and always get the same error:

![error](http://screenshots.vauxoo.com/tulio/10520906718-GUsgtVTjUY.jpg)

if the container have no port opened to the world it works just fine:

![good](http://screenshots.vauxoo.com/tulio/10531206718-Uy4zgIuIVm.jpg)

There is no service in the host using such ports (these are the ones I tried):

![ports](http://screenshots.vauxoo.com/tulio/10540606718-kJ8gUXp5JX.jpg)

Also checked the endpoint via the api for that particular host:

![endpoit](http://screenshots.vauxoo.com/tulio/10552006718-qmVG2hSH3a.jpg)

Actually I executed the docker-compose.yml and it worked just fine.

In the environment is only that stack (and the default created by rancher)

is there any other information I can give you to help here?

Regards.
